edge_node.go

```diff
@@ -24,21 +25,73 @@ type edgeNode struct {
 	lastOnlineAt        time.Time
 	lastOfflineAt       time.Time
 	birthSequenceNumber int64
+	devices             map[string]device
```

Comment:
What's this for? Is the library stateful?

Reply:
yeap, it needs to keep state as part of the spec. Both to determine when nodes/devices come online/offline as well as for guaranteeing that messages are processed in order even when they might arrive out of order from the broker (as it will be the case for AWS IoT)

Reply:
Yes. It is. I just got h how this is used